### PR TITLE
Add com.linkedin.gobblin to alias resolver

### PR DESCRIPTION
**Pull request by pcadabam**
@ibuenros can you review?

### CODE REVIEW

File: gobblin-api/src/test/java/com/linkedin/gobblin/TestAlias.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
```
Comment (abti) [inline gobblin-api/src/test/java/com/linkedin/gobblin/TestAlias.java]: Please change to use the Apache 2.0 License header without copyright

File: gobblin-api/src/main/java/gobblin/util/ClassAliasResolver.java [hunk 2]
```diff
@@ -38,16 +39,17 @@
...
  * </b>
  */
 @Slf4j
 public class ClassAliasResolver<T> {
 
-  // Scan all packages in the classpath with prefix gobblin when class is loaded.
+  // Scan all packages in the classpath with prefix gobblin, com.linkedin.gobblin when class is loaded.
   // Since scan is expensive we do it only once when class is loaded.
-  private static final Reflections REFLECTIONS = new Reflections("gobblin");
+  private static final Reflections REFLECTIONS = new Reflections(new ConfigurationBuilder().forPackages("gobblin",
```
Comment (abti) [inline gobblin-api/src/main/java/gobblin/util/ClassAliasResolver.java]: This should be configurable. Also, why are we scanning classes within com.linkedin.gobblin package, when the packages are gobblin.* in our code base?

Comment (pcadabam) [inline gobblin-api/src/main/java/gobblin/util/ClassAliasResolver.java]: Linkedin internal packages are `com.linkedin.gobblin`

**Comment by kadaan** [comment]
It would be ideal if the classpath scanner package list could be configured.  Trying to add a new implementation of a class that is loaded via Alias doesn't work currently unless I put it within the gobblin package.  The same is true when trying to use this in another context.

**Comment by abti** [comment]
https://issues.apache.org/jira/browse/GOBBLIN-65

Please update your PR title with following prefix: [GOBBLIN-65]

**Pull request closed without merging**